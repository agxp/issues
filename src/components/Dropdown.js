import React, {Component} from 'react';

export default class Dropdown extends Component {
    render() {
        const id = this.props.id;
        const name = this.props.name;
        const data = this.props.data;

        if (id != null) {
            return (

                <div className="dropdown float-right">
                    <button className="btn btn-secondary dropdown-toggle" type="button" id={id}
                            data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        {name}
                    </button>
                    <div className="dropdown-menu" aria-labelledby={id}>
                        {data}
                    </div>
                </div>

            );
        }
        return null

    }
}
