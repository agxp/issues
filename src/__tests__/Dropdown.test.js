import React from 'react';
import Dropdown from '../components/Dropdown';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

test('Dropdown should render correctly', () => {
    const dropdown = shallow(<Dropdown id="authorDropdown" />);

    expect(dropdown).toMatchSnapshot();
});

test('Dropdown has proper id', () => {
    const dropdown = shallow(<Dropdown id="authorDropdown" />);

    expect(dropdown.find("#authorDropdown").length).toBe(1)
});

test('Dropdown has proper number elements', () => {
    const n = [1, 2, 3];

    const authors = n.map(a => <a key={a}>{a}<br/></a>);
    const dropdown = shallow(<Dropdown id="authorDropdown" data={authors} />);

    expect(dropdown.find(".dropdown-menu a").length).toBe(3)
});