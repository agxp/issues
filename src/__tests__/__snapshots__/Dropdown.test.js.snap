// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown
    id="authorDropdown"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          aria-expanded="false"
          aria-haspopup="true"
          className="btn btn-secondary dropdown-toggle"
          data-toggle="dropdown"
          id="authorDropdown"
          type="button"
/>,
        <div
          aria-labelledby="authorDropdown"
          className="dropdown-menu"
/>,
      ],
      "className": "dropdown float-right",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-expanded": "false",
          "aria-haspopup": "true",
          "children": undefined,
          "className": "btn btn-secondary dropdown-toggle",
          "data-toggle": "dropdown",
          "id": "authorDropdown",
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-labelledby": "authorDropdown",
          "children": undefined,
          "className": "dropdown-menu",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            aria-expanded="false"
            aria-haspopup="true"
            className="btn btn-secondary dropdown-toggle"
            data-toggle="dropdown"
            id="authorDropdown"
            type="button"
/>,
          <div
            aria-labelledby="authorDropdown"
            className="dropdown-menu"
/>,
        ],
        "className": "dropdown float-right",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-expanded": "false",
            "aria-haspopup": "true",
            "children": undefined,
            "className": "btn btn-secondary dropdown-toggle",
            "data-toggle": "dropdown",
            "id": "authorDropdown",
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-labelledby": "authorDropdown",
            "children": undefined,
            "className": "dropdown-menu",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
